modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -300,6 +303,15 @@ function AddFDSFilterModalContent({
 		);
 	};
 
+	useEffect(() => {
+		if (preselectedValues?.length === 0) {
+			setIncludeMode('include');
+		}
+		else {
+			setIncludeMode(getFilterIncludeMode());
+		}
+	}, [preselectedValues, getFilterIncludeMode]);
+
```

Comment:
`useCallback` is typically a strong indicator that code can be refactored in a cleaner way. It is effective only when we are intentionally memoizing, which is not the case here.

I don't think we need this `useEffect`. Instead, could we extend [this onPreselectedValuesChange](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx#L403) to do something like
```js
onPreselectedValuesChange={(values) => {
	setPreselectedValues(values);

	setIncludeMode(values?.length ? getFilterIncludeMode() : 'include');
}}
```

Reply:
I'd also love to avoid it, tbh I was more interested in the logic itself. We probably don't want to memoize but that's needed to create the effect dependency.

I'm still in doubt wether this can be just done `onPreselectedValuesChange` (i.e. not in the initial rendering), I'll try, if I'm not able today I'll have to delegate :)

Reply:
If it needs to be in initial rendering, you can set in `useState`. Something like
```js
const [includeMode, setIncludeMode] = useState<string>(
	preselectedValues?.length ? getFilterIncludeMode() : 'include'
);
```